### ISSUE TITLE
Argument list too long [ln]

### ISSUE THREAD

**Issue by prodrigestivill** [prodrigestivill/docker-postgres-backup-local#157]
### Discussed in https://github.com/prodrigestivill/docker-postgres-backup-local/discussions/156

<div type='discussions-op-text'>

<sup>Originally posted by **Thorlake** September  2, 2024</sup>
Hi guys,
Sorry for bothering
By any chance is there a way to help me?

For version
```
prodrigestivill/postgres-backup-local:14
```
and configuration 
```
  POSTGRES_EXTRA_OPTS: -Fd -Z9 --schema=public -b
  SCHEDULE: 0 12 * * *
  BACKUP_KEEP_DAYS: 9
  BACKUP_KEEP_WEEKS: 0
  BACKUP_KEEP_MONTHS: 3
  BACKUP_KEEP_MINS: 0
```
I got the exception
```
./backup.sh: line 105: /usr/bin/ln: Argument list too long
```
I assume the code from https://github.com/prodrigestivill/docker-postgres-backup-local/blob/main/backup.sh#L105
```
ln -f "${FILE}/"* "${DFILENEW}/"
```
just links folder `/last` to currently created folder. However there are maybe too many files and `ln` can't be executed

file name is usually something like
```
/backups/daily/superuser_db-20240822.sql.gz/3419.dat.gz
/backups/daily/superuser_db-20240822.sql.gz/3420.dat.gz
/backups/daily/superuser_db-20240822.sql.gz/3431.dat.gz
....
```

What is the best way to fix it?
To fix project code to sth like
````
for f in "${FILE}"/*; do
  ln -f "$f" "${DFILENEW}/"
done
```
same for WFILENEW and MWFILENEW

upd: checked and I do have really a big number of files like `-20240902-100510.sql.gz/blob_17240.dat.gz'`
because number of blob files is huge (with directory custom dump type)
</div>

**Comment by prodrigestivill** [comment]
Uploaded your proposed fix, lets hope that for doesn't have the same max argument list.
Otherwise we can switch to find or similar.

@Thorlake Can you test with the latest version?

**Comment by prodrigestivill** [comment]
Yes, test failed since the CI where using relative `BACKUP_DIR` variable which doesn't make sense as per a real docker-compose configuration.

Fixed in 2cbc34e8e6bb4478b98d7c8110b6a7076a136ded.

Also more changes have been made like always validate environment variables before starting go-cron, as well as allow initial backup on start, etc...

**Comment by prodrigestivill** [comment]
Closing, please reopen if this is not yet fixed.
